src/redisai.c

```diff
@@ -1826,104 +900,18 @@ int RedisAI_Config_RedisCommand(RedisModuleCtx *ctx, RedisModuleString **argv, i
   return RedisModule_ReplyWithError(ctx, "ERR unsupported subcommand");
 }
 
-/** 
-* DAGRUN Building Block to parse [LOAD <nkeys> key1 key2... ]
-*/
-int RAI_parseDAGLoadArgs(RedisModuleCtx *ctx, RedisModuleString **argv, int argc, AI_dict **localContextDict,
-                              const char* chaining_operator) {
-  if (argc < 3) {
-    RedisModule_WrongArity(ctx);
-    return -1;
-  }
-
-  long long n_keys;
-  const int retval = RedisModule_StringToLongLong(argv[1], &n_keys);
-  if (retval != REDISMODULE_OK || n_keys <= 0) {
-    RedisModule_ReplyWithError(
-        ctx, "ERR invalid or negative value found in number of keys to LOAD");
-    return -1;
-  }
-
-  int number_loaded_keys = 0;
-  int separator_flag = 0;
-  size_t argpos = 2;
-  for (; (argpos <= argc - 1) && (number_loaded_keys < n_keys);
-       argpos++) {
-    const char *arg_string = RedisModule_StringPtrLen(argv[argpos], NULL);
-    if (!strcasecmp(arg_string, chaining_operator)) {
-      separator_flag = 1;
-      break;
-    } else {
-      RAI_Tensor *t;
-      RedisModuleKey *key;
-      const int status =
-          RAI_GetTensorFromKeyspace(ctx, argv[argpos], &key, &t, REDISMODULE_READ);
-      if (status == REDISMODULE_ERR) {
-        RedisModule_Log(ctx, "warning", "on DAGRUN's LOAD could not load tensor %s from keyspace", arg_string);
-        return -1;
-      }
-      AI_dictAdd(*localContextDict, arg_string, t);
-      number_loaded_keys++;
-    }
-  }
-  if(number_loaded_keys != n_keys){
-    RedisModule_WrongArity(ctx);
-    return -1;
-  }
-  return argpos;
-}
-
-/** 
-* DAGRUN Building Block to parse [PERSIST <nkeys> key1 key2... ]
-*/
-int RAI_parseDAGPersistArgs(RedisModuleCtx *ctx, RedisModuleString **argv,
-                         int argc, AI_dict **localContextDict,
-                         const char *chaining_operator) {
-  if (argc < 3) {
-    RedisModule_WrongArity(ctx);
-    return -1;
-  }
-
-  long long n_keys;
-  const int retval = RedisModule_StringToLongLong(argv[1], &n_keys);
-  if (retval != REDISMODULE_OK || n_keys <= 0) {
-    RedisModule_ReplyWithError(
-        ctx, "ERR invalid or negative value found in number of keys to LOAD");
-    return -1;
-  }
-
-  int number_loaded_keys = 0;
-  int separator_flag = 0;
-  size_t argpos = 2;
-  for (; (argpos <= argc - 1) && (number_loaded_keys < n_keys); argpos++) {
-    const char *arg_string = RedisModule_StringPtrLen(argv[argpos], NULL);
-    if (!strcasecmp(arg_string, chaining_operator)) {
-      separator_flag = 1;
-      break;
-    } else {
-      AI_dictAdd(*localContextDict, arg_string, 1);
-      number_loaded_keys++;
-    }
-  }
-  if (number_loaded_keys != n_keys) {
-    RedisModule_WrongArity(ctx);
-    return -1;
-  }
-  return argpos;
-}
-
 /**
- * AI.DAGRUN [LOAD <nkeys> key1 key2... ] [PERSIST <nkeys> key1 key2... ] |>
+ * AI.DAGRUN device [LOAD <nkeys> key1 key2... ] [PERSIST <nkeys> key1 key2... ] |>
```

Comment:
I don't think we should have a DAG-wise device. This limits DAG in the future.
If we limit the current implementation to an individual device (I imagine thats' because this way everything is in the same thread), let's detect it and thow a "non supported yet" error.

Reply:
@lantiga we need a device to add to the queue. I believe we're bound to this

Reply:
As you've mentioned @lantiga bouding ourselfs to parsing the device string would block further use-cases. For now doing as we've discussed and getting the device string from the model info via parsing. If multiple devices are present on the same DAG reply with error for now

Reply:
solved in ab37bbe7320c8b588639afd24645825d06e89f4d